homeassistant/components/sensor/__init__.py

```diff
@@ -508,67 +568,62 @@ def state(self) -> Any:
                 )
             return value
 
+        numerical_value: int | float | Decimal | None = None
+        numerical_conversion_failed = None
+        if not isinstance(value, (int, float, Decimal)):
+            try:
+                numerical_value = float(value)  # type:ignore[arg-type]
+            except (TypeError, ValueError):
+                numerical_conversion_failed = True
+        else:
+            numerical_value = value
```

Comment:
With this code, integer strings will be converted to floats.
eg. `"1310"` would be converted to `1310.0`

Could we keep them as integer?